### ISSUE TITLE
Abort Trap 6 on OSX

### ISSUE THREAD

**Issue by lalaithion** [hashcat/hashcat#309]
I'm running a built-from-source version of oclHashcat on a Macbook Pro with a NVidia GeForce GT 750M GPU. It runs the benchmark (oclHashcat -b) fine, as well as all three of the example shell scripts. 

However, when run with the -m 1800 flag (aka sha512crypt $6$, SHA512(Unix)), it gives the following output:

`Device #1: Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz, skipped`
`Device #2: Iris Pro, 384/1536 MB allocatable, 1200Mhz, 40MCU`
`Device #3: GeForce GT 750M, 512/2048 MB allocatable, 925Mhz, 2MCU`

`Device #2: Kernel` `/Users/izaakweiss/Desktop/Programming/oclHashcat/./kernels/m01800.9d03d613.kernel (663708 bytes)`
`Device #2: Kernel` `/Users/izaakweiss/Desktop/Programming/oclHashcat/./kernels/amp_a0.9d03d613.kernel (141323 bytes)`
`Device #3: Kernel` `/Users/izaakweiss/Desktop/Programming/oclHashcat/./kernels/m01800.e70881bf.kernel (528033 bytes)`
`Device #3: Kernel` `/Users/izaakweiss/Desktop/Programming/oclHashcat/./kernels/amp_a0.e70881bf.kernel (81815 bytes)`

`Checking for weak hashes...Abort trap: 6`

**Comment by matrix** [comment]
Hi @lalaithion,
probably the problem it's the device 3.
Try with -d 2 or -d 1,2 to filter-out the device 3 and tell me if work.

Bye,
matrix

**Comment by lalaithion** [comment]
You were close; the problem turned out to be device 2. Thanks.

**Comment by jsteube** [comment]
I'm pretty sure other user have the same problem then. We need to fix this before other users went into the same problem. My first try to fix would be to edit the tuning database. Please edit hashcat_tuning.hctab.
- Find the line
  Iris                                            \*       1800    1       1       16

(Note github throws out some spaces here, I hope it's clear what I mean)

Try to set the '16' value to '1' and try if the problem still occurs

**Comment by jsteube** [comment]
@gm4tr1x OP is not responding. Can you test please?

**Comment by lalaithion** [comment]
Hey, sorry for no response.

I changed that line, and I'm still getting the same error.

**Comment by matrix** [comment]
@lalaithion could you describe how you test it? Write here the cmdline please.
Thanks,
matrix

**Comment by matrix** [comment]
My Iris gpu card work (Iris, 384/1536 MB allocatable, 1200Mhz, 40MCU)

http://pastebin.com/fWqsKuA5

@lalaithion @jsteube

**Comment by jsteube** [comment]
@gm4tr1x Do you think we can close this issue?

**Comment by jsteube** [comment]
Problem fixed

**Comment by hubert3** [comment]
Still happening for me, new MacBook Pro 15" (2016), hashcat compiled from Github code today

It runs fine without -r but crashes with any larger rules file I try.

16 hashes in the file

```
$ ./hashcat -m 5600 NTLMv2.txt ~/combined/combined_unique.txt -r rules/InsidePro-HashManager.rule
hashcat (v3.6.0-455-g8fb583f0) starting...

OpenCL Platform #1: Apple
=========================
* Device #1: Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz, skipped.
* Device #2: Intel(R) HD Graphics 530, 384/1536 MB allocatable, 24MCU
* Device #3: AMD Radeon Pro 455 Compute Engine, 512/2048 MB allocatable, 12MCU

Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 16: z
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 17: Z
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1402: Sa@
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1403: So0
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1404: Si1
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1405: Su4
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1406: Se3
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1408: Si!
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1409: Ss5
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1410: Ss$
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1411: Sl1
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1412: Sg9
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1413: St7
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1414: Sb5
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1415: Sb8
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1416: Ss3
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1417: Sa4
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1419: Sa@So0
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1420: Sa@Si1
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1421: Sa@Su4
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1422: Sa@Se3
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1423: So0Si1
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1424: So0Su4
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1425: So0Se3
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1426: Si1Su4
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1427: Si1Se3
Skipping invalid or unsupported rule in file rules/InsidePro-HashManager.rule on line 1428: Su4Se3
Hashes: 16 digests; 16 unique digests, 16 unique salts
Bitmaps: 16 bits, 65536 entries, 0x0000ffff mask, 262144 bytes, 5/13 rotates
Rules: 6470

Applicable optimizers:
* Zero-Byte
* Not-Iterated

Password length minimum: 0
Password length maximum: 256

Watchdog: Temperature abort trigger disabled.
Watchdog: Temperature retain trigger disabled.

INFO: Removed 2 hashes found in potfile.

Dictionary cache hit:
* Filename..: /Users/hubert/combined/combined_unique.txt
* Passwords.: 26897522
* Bytes.....: 271713531
* Keyspace..: 174026967340





[s]tatus [p]ause [r]esume [b]ypass [c]heckpoint [q]uit =>



[s]tatus [p]ause [r]esume [b]ypass [c]heckpoint [q]uit =>



[s]tatus [p]ause [r]esume [b]ypass [c]heckpoint [q]uit =>



[s]tatus [p]ause [r]esume [b]ypass [c]heckpoint [q]uit => Abort trap: 6
```

**Comment by hubert3** [comment]
Problem in my case is device 2 (Intel(R) HD Graphics 530, 384/1536 MB allocatable, 24MCU)

No crash if I run with -d 3 only

**Comment by hubert3** [comment]
Workaround is -D 1 (use CPU only)

Or -d 2 -u 1 --force (use Intel GPU only with --kernel-loops 1, reduced performance)

Higher -u values crash after a while on my system (5 after a few hours, 7+ soon after launch, 10 immediately on launch)

The crash also does not happen with -d 3 (selecting only the AMD Radeon Pro 455 device), however as per https://github.com/hashcat/hashcat/issues/1350, this is currently not a valid workaround as support for that AMD device seems to be broken. It looks like it's running normally but never finds any passwords.

**TL;DR** Hashcat is not usable for full-speed GPU cracking on OSX right now, at least not on recent hardware like MacBook Pro (15-inch, 2016) running macOS Sierra 10.12.6 with the GPUs shown above.

**Comment by hubert3** [comment]
-d 2 -u 5 using this Intel device:

Intel(R) HD Graphics 530, 384/1536 MB allocatable, 24MCU

Crashes with Abort Trap 6 after a few hours.

It seems that Hashcat is basically not usable for full-speed GPU cracking on OSX at present.

**Comment by MahatmaCrond** [comment]
> Workaround is -D 1 (use CPU only)
> 
> Or -d 2 -u 1 --force (use Intel GPU only with --kernel-loops 1, reduced performance)
> 
> Higher -u values crash after a while on my system (5 after a few hours, 7+ soon after launch, 10 immediately on launch)
> 
> The crash also does not happen with -d 3 (selecting only the AMD Radeon Pro 455 device), however as per #1350, this is currently not a valid workaround as support for that AMD device seems to be broken. It looks like it's running normally but never finds any passwords.
> 
> **TL;DR** Hashcat is not usable for full-speed GPU cracking on OSX right now, at least not on recent hardware like MacBook Pro (15-inch, 2016) running macOS Sierra 10.12.6 with the GPUs shown above.

Thanks for this. Does anyone know if this has been addressed yet or if there is support yet?
